### ISSUE TITLE
Person Record Index is indexing on space to left of comma

### ISSUE THREAD

**Issue by PatientRowan** [historyforge/historyforge#131]
I have long noticed and been puzzled by the null/blank space that is sometimes supplied after the Last Name but before the comma in results indexes.  I don’t know how they happened; it’s not an actual blank character space in the census record, as far as I can tell.
     The Person Record index is indexing on the blank space. The three Besemer-space-commas index after the other Besemers without a space:
![2F766FA7-7E8F-4F02-9B52-B4F93772ADFD](https://user-images.githubusercontent.com/114665933/199766819-21b3f076-c8b6-4ed2-8e2c-80d2685c30f5.jpeg)
       I was able to resolve Emma and Martin by merging them into records higher in the index without spaces, but I couldn’t figure out Burhans: so he’s still there, looking out of place.
      It’s been a few weeks since we’ve been working in the person records, and this is the first time I’ve seen those spaces before commas.  Whereas my memory of them in census records is that they are more common.  It seems to affect matching as well as indexing, at least in these 2 cases.  Do you have a sense of how common they are?  Can we get rid of them programmatically, or is it just not a big issue?

**Comment by dfurber** [comment]
Somehow Burhan's first name had a trailing space. Not sure how it got there. The census records and buildings are set to auto strip such spaces, but person records are not. I've changed that to auto strip extra spaces from person name and pob fields. Right now it is programmatically updating the person records so you don't have to do it manually.

While watching that process unfold locally, I noticed that most of the changes when saving the person record are to the estimated birth year. That must be because it estimates time and place of birth when saving the person record, but that connecting a census record to a person record didn't trigger a re-estimation. I updated the code so that when you add or remove a person record from a census record, it'll auto save the person record to account for the new situation.

Please close this issue when you are satisfied, or reply with comments.

**Comment by dfurber** [comment]
My guess as to how Burhans got that way is that the person record got generated from the census record before the auto strip happened. While that will still be the case, the person record will do its own string stripping.

**Comment by PatientRowan** [comment]
I used to see those spaces in the census record indexes all the time.  Your auto strip explains why I stopped seeing them.  That’s very good.
I had noticed that the birth year did not re-calculate once it was set.  When I review records, I sometimes need to change the age of the person on the census record to match what was actually on the census—and have to remember to adjust the person record as well, if it has already been generated.  Let me play with this for a few days to make sure I understand it.
Just to check—if we unclick “Estimated” for birth year because we’ve found a more convincing year in Find-a-Grave (which granted can also be incorrect, but still)—that prevents any re-estimate of birth year, right?  People are amazingly flaky about birth years across censuses.
